client/src/app/project/[projectid]/searchComponents/SortFilterModal.tsx

```diff
@@ -0,0 +1,95 @@
+import { NoteData } from "@/util/models/ProjectModels";
+import { Dispatch, FC, SetStateAction, useState } from "react";
+
+interface Props {
+  noteData: NoteData[];
+  setLocationIDArray: Dispatch<SetStateAction<string []>>;
+  setSortValue: Dispatch<SetStateAction<string>>;
+  sortValue: string;
+}
+
+const SortFilterModal:FC<Props> = ({
+  noteData,
+  setLocationIDArray,
+  setSortValue,
+  sortValue
+}) => {
+  const [expandModal, setExpandModal] = useState(false);
+  const [filterValues, setFilterValues] = useState<string []>([]);
+  const [lastFilterValues, setLastFilterValues] = useState<string []>([]);
+
+  //update filterValues on change of filter checkboxes
+  const handleFilterChange = (newVal: string) => {
+    if (filterValues.includes(newVal)) {
+      setFilterValues((curr) => curr.filter((val) => val != newVal));
+    } else {
+      setFilterValues([...filterValues, newVal]);
+    }
+  };
+
+  //add location IDs of filtered data to be later passed on to map and search modules
+  const applyFilter = () => {
+    const tempIDs: string[] = [];
+
+    if (filterValues.length != 0) {
+      const filteredData = noteData.filter((note) => filterValues.includes(note.priority));
+      filteredData.forEach((note)=>{tempIDs.push(note.locationID)});
+      setLocationIDArray(tempIDs);
+    } else {
+      setLocationIDArray([]);
+    }
+    setLastFilterValues([...filterValues]);
+    setExpandModal(false);
+  };
+
+  //close modal and set filterValues to last set chosen by user
+  const cancelFilter = () => {
+    setExpandModal(false);
+    setFilterValues([...lastFilterValues]);
+  };
+
+  return (
+    <div>
+      <div className="flex flex-row justify-center items-center pb-4 text-2xl">
+        <div className="pr-4 text-lg font-bold">Searched Places</div>
+      </div>
+
+      <button onClick={() => {
+        setExpandModal(!expandModal);
+      }}>Sort/Filter</button>
+
+      {expandModal ? (
+        <div className="h-full flex justify-center items-center bg-gray-600/50 absolute inset-0">
+          <div className="flex justify-center flex-col items-center rounded-lg bg-Background_Lighter py-5 px-20 relative z-50">
+            <p className="font-medium text-2xl mb-4">Sort</p>
+            <ul className="flex flex-col space-y-2">
+              <li><button onClick={() => setSortValue("name")} className="w-full" autoFocus={sortValue === "name"}>by Name</button></li>
+              <li><button onClick={() => setSortValue("date")} className="w-full" autoFocus={sortValue === "date"}>by Date</button></li>
+              <li><button onClick={() => setSortValue("priority")} className="w-full" autoFocus={sortValue === "priority"}>by Priority</button></li>
+            </ul>
```

Comment:
The UI felt a bit clunky. As discussed, can the sorting be converted to how Spotify does it in like a drop-down menu?

Example: 
![image](https://github.com/AndrewLin95/next_destination_tracker/assets/107276941/1f9627ce-5749-4ed0-a628-eddd1e23b80d)